Modules/Messages/test/Messages.Domain.Tests/Messages/DecomposeForTests.cs

```diff
@@ -0,0 +1,180 @@
+        Add(new TestInput(22, Participant.Sender, Participant.Recipient2, true, true, false, false), new TestOutput(true, true, true, false, false, true, false));
+        Add(new TestInput(23, Participant.Recipient2, Participant.Recipient2, true, true, false, false), new TestOutput(true, true, false, true, false, true, false));
+        Add(new TestInput(24, Participant.Sender, Participant.Recipient1, false, true, true, false), new TestOutput(true, true, true, false, false, true, false));
+        Add(new TestInput(27, Participant.Recipient2, Participant.Recipient2, false, true, true, false), new TestOutput(false, true, true, true, false, false, true));
+        Add(new TestInput(31, Participant.Recipient2, Participant.Recipient2, true, true, true, false), new TestOutput(true, true, true, true, true, true, true));
+        Add(new TestInput(32, Participant.Sender, Participant.Recipient1, false, false, false, true), new TestOutput(true, false, false, true, false, false, false));
+        Add(new TestInput(33, Participant.Recipient1, Participant.Recipient1, false, false, false, true), new TestOutput(false, true, false, true, false, false, false));
+        Add(new TestInput(34, Participant.Sender, Participant.Recipient2, false, false, false, true), new TestOutput(false, false, true, true, false, false, true));
+        Add(new TestInput(37, Participant.Recipient1, Participant.Recipient1, true, false, false, true), new TestOutput(true, true, false, true, false, true, false));
+        Add(new TestInput(38, Participant.Sender, Participant.Recipient2, true, false, false, true), new TestOutput(true, false, true, true, false, false, true));
+        Add(new TestInput(40, Participant.Sender, Participant.Recipient1, false, false, true, true), new TestOutput(true, false, true, true, false, false, true));
+        Add(new TestInput(41, Participant.Recipient1, Participant.Recipient1, false, false, true, true), new TestOutput(false, true, true, true, false, false, true));
+        Add(new TestInput(45, Participant.Recipient1, Participant.Recipient1, true, false, true, true), new TestOutput(true, true, true, true, true, true, true));
+        Add(new TestInput(48, Participant.Sender, Participant.Recipient1, false, true, false, true), new TestOutput(true, true, false, true, false, true, false));
+        Add(new TestInput(50, Participant.Sender, Participant.Recipient2, false, true, false, true), new TestOutput(false, true, true, true, false, false, true));
+        Add(new TestInput(54, Participant.Sender, Participant.Recipient2, true, true, false, true), new TestOutput(true, true, true, true, true, true, true));
+        Add(new TestInput(56, Participant.Sender, Participant.Recipient1, false, true, true, true), new TestOutput(true, true, true, true, true, true, true));
+        // Add(new TestInput(4, Participant.Sender, Participant.Recipient1, true, false, false, false), new TestOutput("#", "#", "#", "#"));
```

Comment:
These cases can _logically_ exist. But they will never happen in the real world. I wasn't sure what to do with them. But at the same time, I didn't want to delete them, because at some point we might decide to consider them in the implementation. And recreating them is a pain.

But yeah, I don't like all that commented out code either :(

Any suggestions?

Reply:
Maybe a single comment that points to where one could find the other test cases and an explanation on why they were not considered?
We can also just keep them (uncommented) since they can, as you stated, logically exist.

Reply:
I uncommented them and added an implementation that throws an exception in such cases.